fix(core): run plugins sequentially

matejchalk opened the pull request:
This fix should prevent problems with concurrent spinners, such as in #1144. 

Plugins are intended to be executed sequentially, but it appears they were (unintentionally?) made concurrent in 7a592ebd268cb9b211da29e3473c4170f9ad4525 as part of the large PR #811.

I intend to make more changes to the plugin execution (log groups, etc.), but for now this is just a minimal fix.

hanna-skryl reviewed (changes requested):
I tested this change, but the concurrent spinner error still occurs:

1. Applied sequential execution in an external repository
2. Modified source code and verified changes took effect
3. Error still thrown during Axe plugin execution

The spinner conflict appears to be caused by something other than parallel plugin execution.

matejchalk commented:
> I tested this change, but the concurrent spinner error still occurs:
> 
> 1. Applied sequential execution in an external repository
> 2. Modified source code and verified changes took effect
> 3. Error still thrown during Axe plugin execution
> 
> The spinner conflict appears to be caused by something other than parallel plugin execution.

Weird, I'm not sure why that would be. Concurrent spinners could be caused by `Promise.all`/`Promise.allSettled` or spinner nesting. Running another spinner after the previous one resolved should work fine. I added some extra tests to the logger to check this, just in case (#1147).

The only spinner usage is in `executeProcess` calls at the moment, so I'm having a hard time figuring out how there can be more than 1 running. The only `executeProcess` I can find in `axe-plugin` is the `playwright install` command that runs for each URL, but in your async `reduce` you await the previous value, so I don't see any possible conflict there. :confused: 

If you'd like me to help debug the problem, I'll probably need to reproduce it. Do you think you could share some minimal reproduction?

The Axe plugin is working in this repo, so I'd start by trying to figure out what's different about the configuration in the repo where you're seeing the error (what other plugins are used, etc.).

How are you applying the changes from this PR, BTW? Editing the downloaded `.js` files in `node_modules/@code-pushup/core`? You could also try installing via `npm i https://pkg.pr.new/code-pushup/cli/@code-pushup/axe-plugin@1145`.

matejchalk commented:
Updating to `0.87.2` fixed the error for me in the portal. :tada: 

<img width="1201" height="739" alt="Screenshot from 2025-11-14 19-08-15" src="https://github.com/user-attachments/assets/af8bb722-31df-4f48-928c-87f4f523e940" />

----

@hanna-skryl All good on your end?
